@xen-orchestra/backups/_runners/_writers/IncrementalXapiWriter.mjs

```diff
@@ -72,18 +72,26 @@ export class IncrementalXapiWriter extends MixinXapiWriter(AbstractIncrementalWr
     this._oldEntries = getOldEntries(settings.copyRetention - 1, listReplicatedVms(xapi, scheduleId, srUuid, vmUuid))
 
     if (settings.deleteFirst) {
-      await this._deleteOldEntries()
+      if (settings.copyRetention > 1) {
+        await this._deleteOldEntries(this._oldEntries)
+        this._oldEntries = []
+      }
+      // avoids deleting base VM
+      else if (this._oldEntries.length > 1) {
+        await this._deleteOldEntries(this._oldEntries.slice(0, -1))
+        this._oldEntries = [this._oldEntries[this._oldEntries.length - 1]]
+      }
```

Comment:
I think this could be easier to follow, and could be even better with a better variable naming, what do you think? 

```suggestion
        // we want to keep the baseVM when copying a delta
        // even if we want to keep only one after
        // @todo : if possible check if it's not a full (full backup interval reached, missing id, ... ) 
        // and prepare should have a parameter isFull 
        // whe can delete safely the old entry if the new one is a full 
        let mostRecentEntry
         if(this._oldEntries.length > 1 && settings.copyRetention === 1){
             mostRecentEntry = this._oldEntries.pop()
         }
         await this._deleteOldEntries() 
         if(mostRecentEntry !==undefined){
            this._oldEntries = [mostRecentEntry]
         } else {
             this._oldEntries = null
         }
```